server/src/main/java/org/elasticsearch/search/aggregations/bucket/geogrid/GeoGridTiler.java

```diff
@@ -189,17 +193,21 @@ public int setValues(GeoShapeCellValues values, MultiGeoValues.GeoValue geoValue
                 return 0;
             }
 
-
             final double tiles = 1 << precision;
             int minXTile = GeoTileUtils.getXTile(bounds.minX(), (long) tiles);
             int minYTile = GeoTileUtils.getYTile(bounds.maxY(), (long) tiles);
             int maxXTile = GeoTileUtils.getXTile(bounds.maxX(), (long) tiles);
             int maxYTile = GeoTileUtils.getYTile(bounds.minY(), (long) tiles);
             int count = (maxXTile - minXTile + 1) * (maxYTile - minYTile + 1);
             if (count == 1) {
-                values.resizeCell(1);
-                values.add(0, GeoTileUtils.longEncodeTiles(precision, minXTile, minYTile));
-                return 1;
+                Rectangle rectangle = GeoTileUtils.toBoundingBox(minXTile, minYTile, precision);
+                GeoRelation relation = geoValue.relate(rectangle);
+                if (relation != GeoRelation.QUERY_DISJOINT) {
```

Comment:
It is unfortunate that we need to build the rectangle even in the Unbounded case.